apps/openmw/mwgui/travelwindow.cpp

```diff
@@ -170,47 +170,18 @@ namespace MWGui
 
         MWBase::Environment::get().getWorld()->setPlayerTraveling(true);
 
-        mTravelTarget = TravelTarget{ *_sender->getUserData<ESM::Position>(), _sender->getUserString("Destination"), _sender->getUserString("interior") == "y" };
+        mTravelTarget = MWBase::World::TravelTarget{ *_sender->getUserData<ESM::Position>(), _sender->getUserString("Destination"), _sender->getUserString("interior") == "y" };
     }
 
-    void TravelWindow::travel()
+    void TravelWindow::onFrame()
     {
         if (!MWBase::Environment::get().getWorld()->isPlayerTraveling())
             return;
-
-        MWWorld::Ptr player = MWMechanics::getPlayer();
-
-        if (!mTravelTarget.interior)
-        {
-            ESM::Position playerPos = player.getRefData().getPosition();
-            float d = (osg::Vec3f(mTravelTarget.pos.pos[0], mTravelTarget.pos.pos[1], 0) - osg::Vec3f(playerPos.pos[0], playerPos.pos[1], 0)).length();
-            int hours = static_cast<int>(d / MWBase::Environment::get().getWorld()->getStore().get<ESM::GameSetting>().find("fTravelTimeMult")->getFloat());
-            for (int i = 0; i < hours; i++)
-            {
-                MWBase::Environment::get().getMechanicsManager()->rest(true);
-            }
-            MWBase::Environment::get().getWorld()->advanceTime(hours);
-        }
-
+        
         MWBase::Environment::get().getWindowManager()->removeGuiMode(GM_Travel);
         MWBase::Environment::get().getDialogueManager()->goodbyeSelected();
 
-        // Teleports any followers, too.
-        MWWorld::ActionTeleport action(mTravelTarget.interior ? mTravelTarget.cellname : "", mTravelTarget.pos, true);
-        action.execute(player);
-
-        MWBase::Environment::get().getWorld()->setPlayerTraveling(false);
-
-        MWBase::Environment::get().getWindowManager()->fadeScreenOut(0);
-        MWBase::Environment::get().getWindowManager()->fadeScreenIn(1);
-    }
-
-    void TravelWindow::onFrame()
-    {
-        if (!MWBase::Environment::get().getWorld()->isPlayerTraveling())
-            return;
-        
-        travel();
+        MWBase::Environment::get().getWorld()->travel(mTravelTarget);
```

Comment:
But this would require storing const CellID&, const ESM::Position& on the WorldImp that would be rather ugly imo. 
Are you sure you want it that way?